src/ResourceManager/Profile/Commands.Profile/Environment/AddAzureRMEnvironment.cs

```diff
@@ -169,15 +171,21 @@ public override void ExecuteCmdlet()
                                 env.Value?.GetEndpoint(AzureEnvironment.Endpoint.ResourceManager)?.ToLowerInvariant(),
                                 GeneralUtilities.EnsureTrailingSlash(ARMEndpoint)?.ToLowerInvariant(), StringComparison.CurrentCultureIgnoreCase));
 
-                        var newEnvironment = new AzureEnvironment { Name = this.Name };
+                        IAzureEnvironment newEnvironment = new AzureEnvironment { Name = this.Name };
+                        var defProfile = DefaultProfile as AzureRmProfile;
```

Comment:
Use thsi method to get the profile: https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/Profile/Commands.Profile/Common/AzureContextModificationCmdlet.cs#L54  and this method: https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/Common/Commands.Common.Authentication.ResourceManager/IProfileOperations.cs#L42 to get the environment by name